### PR TITLE
Convert remaining transcoding tests to opaque pointers

### PR DESCRIPTION
**Pull request by svenvh**
This converts the remaining transcoding tests that weren't converted in previous PRs.

### CODE REVIEW

File: test/transcoding/OpPhi_ArgumentsPlaceholders.ll [hunk 1]
```diff
@@ -23,16 +23,16 @@
 target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
 target triple = "spir"
 
-%struct.Node = type { %struct.Node.0 addrspace(1)* }
+%struct.Node = type { ptr addrspace(1) }
```
Comment (vmaksimo) [inline test/transcoding/OpPhi_ArgumentsPlaceholders.ll:26]: Is `%struct.Node.0` really unneeded here? then looks like the line below can be removed, if I understand it right

File: test/transcoding/annotate_attribute.ll [hunk 2]
```diff
@@ -110,64 +110,64 @@ entry:
...
-  %f4 = getelementptr inbounds %struct.bar, %struct.bar* %s1, i32 0, i32 3
-  %6 = call i8* @llvm.ptr.annotation.p0i8.p0i8(i8* %f4, i8* getelementptr inbounds ([22 x i8], [22 x i8]* @.str.7, i32 0, i32 0), i8* getelementptr inbounds ([23 x i8], [23 x i8]* @.str.1, i32 0, i32 0), i32 9, i8* bitcast ({ i32, i32, i32 }* @.args.1 to i8*))
-  store i8 0, i8* %6, align 4, !tbaa !13
-  %7 = bitcast %struct.bar* %s1 to i8*
-  call void @llvm.lifetime.end.p0i8(i64 12, i8* %7) #4
+  %f4 = getelementptr inbounds %struct.bar, ptr %s1, i32 0, i32 3
+  %6 = call ptr @llvm.ptr.annotation.p0.p0(ptr %f4, ptr @.str.7, ptr @.str.1, i32 9, ptr @.args.1)
+  store i8 0, ptr %6, align 4, !tbaa !13
+  %7 = bitcast ptr %s1 to ptr
+  call void @llvm.lifetime.end.p0(i64 12, ptr %7) #4
```
Comment (vmaksimo) [inline test/transcoding/annotate_attribute.ll:170]: Can it be simplified to ?
```suggestion
  call void @llvm.lifetime.end.p0(i64 12, ptr %s1) #4
```

File: test/transcoding/CreatePipeFromPipeStorage.ll [hunk 3]
```diff
@@ -36,13 +36,13 @@ target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:2
 target triple = "spir-unknown-unknown"
 
 %spirv.ConstantPipeStorage = type { i32, i32, i32 }
-%"class.cl::pipe_storage<int __attribute__((ext_vector_type(4))), 1>" = type { %spirv.PipeStorage addrspace(1)* }
+%"class.cl::pipe_storage<int __attribute__((ext_vector_type(4))), 1>" = type { ptr addrspace(1) }
```
Comment (vmaksimo) [inline test/transcoding/CreatePipeFromPipeStorage.ll:39]: Same, is `%spirv.PipeStorage` really unneeded anymore?

Comment (svenvh) [reply test/transcoding/CreatePipeFromPipeStorage.ll:39]: I've removed it.

File: test/transcoding/annotate_attribute.ll (hunk 2 above)
Comment (svenvh) [reply test/transcoding/annotate_attribute.ll]: Yes; I've updated this and other occurrences.

File: test/transcoding/OpPhi_ArgumentsPlaceholders.ll (hunk 1 above)
Comment (svenvh) [reply test/transcoding/OpPhi_ArgumentsPlaceholders.ll:26]: Good catch, I've removed it.

**Review by vmaksimo — APPROVED** [review]

**Review by MrSidims — APPROVED** [review]

**Pull request merged**